server/router_utils.go

```diff
@@ -131,14 +131,16 @@ func readOrganizationID(writer http.ResponseWriter, request *http.Request, auth
 	return types.OrgID(organizationID), err
 }
 
+// checkPermissions checks if the provided organization ID matches with the one of the authenticated user
+// when the authentication is enabled
 func checkPermissions(writer http.ResponseWriter, request *http.Request, orgID types.OrgID, auth bool) error {
 	identityContext := request.Context().Value(types.ContextKeyUser)
 	if identityContext != nil && auth {
 		identity := identityContext.(types.Identity)
 		if identity.Internal.OrgID != orgID {
 			const message = "You have no permissions to get or change info about this organization"
 			log.Error().Msg(message)
-			handleServerError(writer, &AuthenticationError{errString: message})
+			handleServerError(writer, &types.ItemNotFoundError{ItemID: orgID})
```

Comment:
Why do you return item not found when it's authentication error?

Reply:
For sake of consistency. In this issue https://github.com/RedHatInsights/insights-results-aggregator/issues/907 I raised that question and we agreed to return 404. This issue https://github.com/RedHatInsights/insights-results-smart-proxy/issues/70 is about the same.

Reply:
@quarckster I commented there that we can return 403 and it doesn't make sense to return 404 when it's obviously an authentication error. Well, anyway, it's not a big deal, some apis return 404 instead of 403 as well.

Reply:
> some apis return 404 instead of 403 as well.

Yes, and we decided just to follow that pattern.